### PR TITLE
Add node selectors to job and task config

### PR DESCRIPTION
**Pull request by annymsMthd**
Signed-off-by: Joshua Benjamin <joshuabnjmn@gmail.com>

### CODE REVIEW

**Comment by annymsMthd** [comment]
Looks like the `make dockerhub_push` should be removed from PRs.

**Comment by anandswaminathan** [comment]
@annymsMthd Just pushed a PR to remove docker_hub push from PRs.

File: pkg/apis/app/v1alpha1/types.go [hunk 1]
```diff
@@ -101,13 +101,15 @@ type JobManagerConfig struct {
 	Environment           EnvironmentConfig           `json:"envConfig"`
 	Replicas              *int32                      `json:"replicas,omitempty"`
 	OffHeapMemoryFraction *float64                    `json:"offHeapMemoryFraction,omitempty"`
+	NodeSelector          map[string]string           `json:"nodeSelector"`
```
Comment (anandswaminathan) [inline pkg/apis/app/v1alpha1/types.go]: Can you add "omitempty"  like above to prevent serialization when empty ?

**Comment by annymsMthd** [comment]
@anandswaminathan should be gtg now

**Comment by mwylde** [comment]
Hey Joshua — can you give a bit more context on why this is needed?

**Comment by annymsMthd** [comment]
Sure thing. In our kubernetes cluster we have different labels for nodes. Since we self host we have dedicated nodes for certain workloads. e.g. ingress, worker, storage, etc.

**Comment by annymsMthd** [comment]
@mwylde @anandswaminathan any reason not to move forward with this? This is a pretty common thing in k8s.

**Pull request merged**